src/main/kotlin/no/nav/familie/dokument/storage/StorageController.kt

```diff
@@ -13,63 +15,55 @@ import org.springframework.http.MediaType
 
     private val secureLogger = LoggerFactory.getLogger("secureLogger")
 
     /// TODO: "bucket"-path brukes ikke ennå. "familievedlegg" brukes alltid
     @PostMapping(path = ["{bucket}"],
                  consumes = [MediaType.MULTIPART_FORM_DATA_VALUE],
                  produces = [MediaType.APPLICATION_JSON_VALUE])
-    fun addAttachment(@PathVariable("bucket")bucket: String,
+    fun addAttachment(@PathVariable("bucket") bucket: String,
                       @RequestParam("file") multipartFile: MultipartFile): ResponseEntity<Map<String, String>> {
 
         if (multipartFile.isEmpty) {
-            log.info("Dokumentet som lastes opp er tomt - size: [{}] ", multipartFile.size)
-            return ResponseEntity.status(HttpStatus.BAD_REQUEST).build()
+            throw InvalidDocumentSize("Dokumentet som lastes opp er tomt - size: [${multipartFile.size}] ")
         }
 
         val bytes = multipartFile.bytes
-        val maxFileSizeInBytes = maxFileSizeInMb*1024*1024
+        val maxFileSizeInBytes = maxFileSizeInMb * 1024 * 1024
         log.debug("Dokument lastet opp med størrelse (bytes): " + bytes.size)
 
         if (bytes.size > maxFileSizeInBytes) {
-            throw IllegalArgumentException(HttpStatus.PAYLOAD_TOO_LARGE.toString())
+            throw InvalidDocumentSize("Dokumentstørrelsen(${bytes.size} bytes) overstiger grensen(${maxFileSizeInMb} mb)")
         }
 
         val directory = contextHolder.hentFnr()
 
         val uuid = UUID.randomUUID().toString()
 
-        try {
-            storage.put(directory, uuid, bytes)
-        } catch (e: RuntimeException) {
-            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build()
-        }
-
-        return ResponseEntity.ok(mapOf("dokumentId" to uuid, "filnavn" to multipartFile.name))
+        storage.put(directory, uuid, bytes)
+        return ResponseEntity.status(HttpStatus.CREATED).body(mapOf("dokumentId" to uuid, "filnavn" to multipartFile.name))
     }
 
     /// TODO: "bucket"-path brukes ikke ennå. "familievedlegg" brukes alltid
     @GetMapping(path = ["{bucket}/{dokumentId}"], produces = [MediaType.APPLICATION_JSON_VALUE])
     fun getAttachment(@PathVariable("bucket") bucket: String,
                       @PathVariable("dokumentId") dokumentId: String): ResponseEntity<Ressurs<ByteArray>> {
-        val directory = contextHolder.hentFnr()
         return try {
+            val directory = contextHolder.hentFnr()
             val data = storage[directory, dokumentId]
             log.debug("Loaded file with {}", data)
-            ResponseEntity.ok(Ressurs.Companion.success(data))
-        } catch (e: RuntimeException) {
-            secureLogger.error("Henting av vedlegg feilet", e)
-            ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(Ressurs.Companion.failure(e.message))
+            ResponseEntity.ok(Ressurs.success(data))
+        } catch (e: GcpDocumentNotFound) {
+            ResponseEntity.noContent().build()
```

Comment:
I'm not sure about this.
In my meaning 204 should not be used with get. But at the same time `AbstractRestClient` will throw an exception if 2xx is not returned.
@charliemidtlyng @henninghaakonsen ?

Reply:
I think bad request here is more correct, since someone is trying to get a attachment that doesn't exist

Reply:
Agree with 404. But maybe it requires the change on the client apps